### PR TITLE
feat(route): update ftchinese

### PR DESCRIPTION
**Pull request by prnake**
## 完整路由地址 / Example for the proposed route(s)

```routes
/ft/chinese
/ft/chinese/news
/ft/chinese/channel-dailyword
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

Bypass paywall.

由于首页信息流不是按时间排序的，保留了所有内容，首次获取容易超时。

### CODE REVIEW

File: lib/v2/ft/utils.js [hunk 1]
```diff
@@ -2,6 +2,67 @@
 const parser = require('@/utils/rss-parser');
 const cheerio = require('cheerio');
 
+const ParseContent = (content, type = 'p') => (content ? `<${type}>${content}</${type}>` : '');
+
+const ProcessJson = (item, json_data) => {
+    item.title = json_data.cheadline;
+    item.author = json_data.cauthor.replace(/<\/?.+?>/g, '');
```
[the remark this answers is not in the document]
Comment (prnake) [reply lib/v2/ft/utils.js:9]: Doesn't seem to be a problem

File: lib/v2/ft/utils.js [hunk 2]
```diff
@@ -2,6 +2,67 @@ const got = require('@/utils/got');
...
+    item.description += '<br>';
+    item.description += ParseContent(`<b>${json_data.eheadline}</b>`);
+    item.description += ParseContent(json_data.elongleadbody);
+    item.description += ParseContent(json_data.ebody);
+
+    return item;
+};
+
+const ProcessWebview = (feed, $, ctx, site, channel) => {
+    if (!feed || !feed.length) {
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:37]: https://deepscan.io/dashboard/#view=project&tid=6244&pid=8135&bid=92448&subview=pull-request&prid=1541102

File: lib/v2/ft/utils.js [hunk 3]
```diff
@@ -2,6 +2,67 @@ const got = require('@/utils/got');
...
 
+const ParseContent = (content, type = 'p') => (content ? `<${type}>${content}</${type}>` : '');
+
+const ProcessJson = (item, json_data) => {
+    item.title = json_data.cheadline;
+    item.author = json_data.cauthor.replace(/<\/?.+?>/g, '');
+    item.pubDate = new Date(Math.max(json_data.pubdate, json_data.last_publish_time) * 1000).toUTCString();
+
+    item.description = '';
+    item.description += ParseContent(`「${json_data.clongleadbody}」`);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:13]: You can use `<blockquote>`

File: lib/v2/ft/utils.js [hunk 4]
```diff
@@ -2,6 +2,67 @@ const got = require('@/utils/got');
...
+        // seem only to be meta data
+        // item.description += ParseContent(json_data.cshortleadbody);
+    }
+    if (json_data.story_pic && json_data.story_pic.smallbutton) {
+        item.description += `<img src="${json_data.story_pic.smallbutton}"/>`;
+    }
+    item.description += ParseContent(json_data.cbody);
+    item.description += '<br>';
+    item.description += ParseContent(`<b>${json_data.eheadline}</b>`);
+    item.description += ParseContent(json_data.elongleadbody);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:30]: This one should be `blockquote`-ed as well

File: lib/v2/ft/utils.js [hunk 5]
```diff
@@ -2,6 +2,67 @@ const got = require('@/utils/got');
...
+        item.enclosure_url = json_data.cshortleadbody;
+        item.enclosure_type = 'audio/mpeg';
+    } else {
+        // seem only to be meta data
+        // item.description += ParseContent(json_data.cshortleadbody);
+    }
+    if (json_data.story_pic && json_data.story_pic.smallbutton) {
+        item.description += `<img src="${json_data.story_pic.smallbutton}"/>`;
+    }
+    item.description += ParseContent(json_data.cbody);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:27]:
```suggestion
    item.description += json_data.cbody;
```

File: lib/v2/ft/utils.js [hunk 6]
```diff
@@ -2,6 +2,67 @@ const got = require('@/utils/got');
...
+    item.description += '<br>';
+    item.description += ParseContent(`<b>${json_data.eheadline}</b>`);
+    item.description += ParseContent(json_data.elongleadbody);
+    item.description += ParseContent(json_data.ebody);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:31]:
```suggestion
    item.description += json_data.ebody;
```

File: lib/v2/ft/utils.js [hunk 7]
```diff
@@ -2,6 +2,67 @@ const got = require('@/utils/got');
...
+    feed.items = $('.items > .item-container')
+        .map((_, item) => {
+            item = $(item);
+            return {
+                title: item.find('h2').text(),
+                description: item.find('.item-lead').text(),
+                link: item.attr('data-type') ? `https://${site}/${item.attr('data-type')}/${item.attr('data-id')}` : `https://${site}${item.find('a').attr('href')}`,
+            };
+        })
+        .get();
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:63]:
```suggestion
        .toArray()
        .map((item) => {
            item = $(item);
            return {
                title: item.find('h2').text(),
                description: item.find('.item-lead').text(),
                link: item.attr('data-type') ? `https://${site}/${item.attr('data-type')}/${item.attr('data-id')}` : `https://${site}${item.find('a').attr('href')}`,
            };
        });
```

File: lib/v2/ft/utils.js [hunk 8]
```diff
@@ -39,41 +100,79 @@ const ProcessFeed = (i, $, link) => {
...
         try {
-            feed = await parser.parseURL(`https://${site}.ftchinese.com/rss/${channel}`);
+            feed = await parser.parseURL(`https://${site}/rss/${channel}`);
         } catch (error) {
-            return {
-                title: `FT 中文网 ${channel} 不存在`,
-                description: `FT 中文网 ${channel} 不存在`,
-            };
+            try {
+                const previewResponse = await got.get(`https://${site}/${channel}?webview=ftcapp&bodyonly=yes&maxB=1&device=iPad`);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:114]:
```suggestion
                const previewResponse = await got(`https://${site}/${channel}?webview=ftcapp&bodyonly=yes&maxB=1&device=iPad`);
```

File: lib/v2/ft/utils.js [hunk 9]
```diff
@@ -39,41 +100,79 @@ const ProcessFeed = (i, $, link) => {
...
+                return {
+                    title: `FT 中文网 ${channel} 不存在`,
+                    description: `FT 中文网 ${channel} 不存在`,
+                };
+            }
         }
     } else {
-        feed = await parser.parseURL(`https://${site}.ftchinese.com/rss/feed`);
+        try {
+            const previewResponse = await got.get(`https://${site}/?webview=ftcapp&bodyonly=yes&maxB=1&device=iPad`);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:126]:
```suggestion
            const previewResponse = await got(`https://${site}/?webview=ftcapp&bodyonly=yes&maxB=1&device=iPad`);
```

File: lib/v2/ft/utils.js [hunk 10]
```diff
@@ -39,41 +100,79 @@ const ProcessFeed = (i, $, link) => {
...
+                        }
+                    }
+                } catch (error) {
+                    json_data = undefined;
                 }
+                if (json_data) {
+                    return ProcessJson(item, json_data);
+                } else {
+                    try {
+                        const response = await got.get(`${item.link}?full=y&archive`);
```
Comment (TonyRL) [inline lib/v2/ft/utils.js:161]:
```suggestion
                        const response = await got(`${item.link}?full=y&archive`);
```

**Comment by DIYgod** [comment]
This PR has been open for too long and will be closed first. If you have any questions, feel free to continue the discussion below.

**Pull request closed without merging**